dff/messengers/telegram.py

```diff
@@ -0,0 +1,179 @@
+            attachment.source = FilePath(file_name)
+        else:
+            raise ValueError(f"For attachment {attachment} title or id is not defined!")
+
+    async def extract_message_from_telegram(self, update: TelegramMessage) -> Message:  # pragma: no cover
+        message = Message()
+        message.attachments = list()
+
+        if update.text is not None:
+            message.text = update.text
+        if update.location is not None:
+            message.attachments += [Location(latitude=update.location.latitude, longitude=update.location.longitude)]
+        if update.contact is not None:
+            message.attachments += [Contact(phone_number=update.contact.phone_number, first_name=update.contact.first_name, last_name=update.contact.last_name)]
+        if update.invoice is not None:
+            message.attachments += [Invoice(title=update.invoice.title, description=update.invoice.description, currency=update.invoice.currency, amount=update.invoice.total_amount)]
+        if update.poll is not None:
+            message.attachments += [Poll(question=update.poll.question, options=[PollOption(text=option.text, votes=option.voter_count) for option in update.poll.options])]
+        if update.audio is not None:
+            message.attachments += [Audio(id=update.audio.file_id, title=update.audio.file_unique_id)]
+        if update.video is not None:
+            message.attachments += [Video(id=update.video.file_id, title=update.video.file_unique_id)]
+        if update.animation is not None:
+            message.attachments += [Animation(id=update.animation.file_id, title=update.animation.file_unique_id)]
+        if len(update.photo) > 0:
+            message.attachments += [Image(id=picture.file_id, title=picture.file_unique_id) for picture in update.photo]
+        if update.document is not None:
+            message.attachments += [Document(id=update.document.file_id, title=update.document.file_unique_id)]
+
+        return message
+
+    def _create_keyboard(self, buttons: Sequence[Sequence[Button]]) -> Optional[InlineKeyboardMarkup]:  # pragma: no cover
+        button_list = None
+        if len(buttons) > 0:
+            button_list = [[InlineKeyboardButton(text=button.text, callback_data=button.data if button.data is not None else button.text) for button in row] for row in buttons]
+        if button_list is None:
+            return None
+        else:
+            return InlineKeyboardMarkup(button_list)
+
+    async def cast_message_to_telegram_and_send(self, bot: ExtBot, chat_id: int, message: Message) -> None:  # pragma: no cover
+        buttons = list()
+        if message.attachments is not None:
+            files = list()
+            for attachment in message.attachments:
+                if isinstance(attachment, Location):
+                    await bot.send_location(chat_id, attachment.latitude, attachment.longitude, reply_markup=self._create_keyboard(buttons))
+                    return
```

Comment:
Do we want to return?
I think the expected behaviour in case `Message` contains both a text and a location is to send both in two tg message instances.

Reply:
Currently we don't have consistency (empty message will not be sent via this interface).

History will still be parsable if we assume no one spams messages (then consequent tg messages from one side are part of a single `Message` instance).